code/__DEFINES/milla.dm

```diff
@@ -0,0 +1,129 @@
+// milla.dm - DM API for milla extension library
+
+// Default automatic MILLA detection.
+// Look for it in the build location first, then in `.`, then in standard places.
+
+/* This comment bypasses grep checks */ /var/__milla
+
+/proc/__detect_milla()
+	if(world.system_type == UNIX)
+		// First check if it's built in the usual place.
+		if(fexists("./milla/target/i686-unknown-linux-gnu/release/libmilla.so"))
+			return __milla = "./milla/target/i686-unknown-linux-gnu/release/libmilla.so"
+		// Then check in the current directory.
+		if(fexists("./libmilla.so"))
+			return __milla = "./libmilla.so"
+		// And elsewhere.
+		return __milla = "libmilla.so"
+	else
+		// First check if it's built in the usual place.
+		if(fexists("./milla/target/i686-pc-windows-msvc/release/milla.dll"))
+			return __milla = "./milla/target/i686-pc-windows-msvc/release/milla.dll"
+		// Then check in the current directory.
+		if(fexists("./milla.dll"))
+			return __milla = "./milla.dll"
+		// And elsewhere.
+		return __milla = "milla.dll"
+
+#define MILLA (__milla || __detect_milla())
+
+#define MILLA_CALL(func, args...) call_ext(MILLA, "byond:[#func]_ffi")(args)
+
+#define REAL_SET_TILE_ATMOS(args...) MILLA_CALL(set_tile_atmos, args)
+
+/proc/set_tile_atmos(x, y, z, blocked_north, blocked_east, blocked_south, blocked_west, atmos_mode, external_temperature, oxygen, carbon_dioxide, nitrogen, toxins, sleeping_agent, agent_b, temperature, innate_heat_capacity)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
```

Comment:
These sanity checks seem useful, but is there any possibility that these might cause some extra overhead that we might not want? Maybe these could be set under a preprocessor directive or something?

Reply:
See my reply to AA.